frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/ResetCurrentBranchToRemoteAction.java

```diff
@@ -0,0 +1,30 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getCurrentBranchNameIfManaged;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import io.vavr.control.Option;
+
+import com.virtuslab.gitmachete.frontend.actions.common.BaseResetBranchToRemoteAction;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class ResetCurrentBranchToRemoteAction extends BaseResetBranchToRemoteAction {
+  @Override
+  protected Option<String> getBranchName(AnActionEvent anActionEvent) {
```

Comment:
I added `ActionAwareBranchNameProvider` in #245. It would be wise to use it here too :slightly_smiling_face: